src/main/scala/tile/BaseTile.scala

```diff
@@ -103,10 +103,24 @@ trait HasNonDiplomaticTileParameters {
     val f = if (tileParams.core.fpu.nonEmpty) "f" else ""
     val d = if (tileParams.core.fpu.nonEmpty && tileParams.core.fpu.get.fLen > 32) "d" else ""
     val c = if (tileParams.core.useCompressed) "c" else ""
-    val b = if (tileParams.core.useBitManip) "b" else ""
     val v = if (tileParams.core.useVector) "v" else ""
-    val x = tileParams.core.customIsaExt.map(s => s"_$s").getOrElse("")
-    s"rv${p(XLen)}$ie$m$a$f$d$c$b$v$x"
+    val multiLetterExt = Seq(
+      (if (tileParams.core.useBitManip) "Zba" else ""),
+      (if (tileParams.core.useBitManip) "Zbb" else ""),
+      (if (tileParams.core.useBitManip) "Zbc" else ""),
+      (if (tileParams.core.hasBitManipCrypto) "Zbkb" else ""),
+      (if (tileParams.core.hasBitManipCrypto) "Zbkc" else ""),
+      (if (tileParams.core.hasBitManipCrypto) "Zbkx" else ""),
+      (if (tileParams.core.useBitManip) "Zbs" else ""),
+      (if (tileParams.core.useCryptoNIST) "Zknd" else ""),
+      (if (tileParams.core.useCryptoNIST) "Zkne" else ""),
+      (if (tileParams.core.useCryptoNIST) "Zknh" else ""),
+      (if (tileParams.core.useCryptoSM) "Zksed" else ""),
+      (if (tileParams.core.useCryptoSM) "Zksh" else ""),
+      tileParams.core.customIsaExt.getOrElse("")
+    )
+    val multiLetterString = multiLetterExt.filter(_ != "").mkString("_")
+    s"rv${p(XLen)}$ie$m$a$f$d$c$v$multiLetterString"
```

Comment:
Shouldn't there be a `_` before `$multiLetterString`?

Reply:
Well that is accepted format as underscores could be added between extensions.

> Underscores “_” may be used to separate ISA extensions to improve readability and to provide disambiguation

The only required underscore is between the multi-letter extensions, and in the spec the example is

> Extensions with the “Z” prefix must be separated from other multi-letter extensions by an underscore, e.g., “RV32IMACZicsr Zifencei”.

Also strangely riscv-software-src/riscv-config accepts rv32imaczicsr but does not accept rv32ima_zicsr.